torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1100,15 +1100,18 @@ std::vector<std::shared_ptr<NCCLComm>>& ProcessGroupNCCL::getNCCLComm(
   // Create the unique NCCL ID and broadcast it
   ncclUniqueId ncclID;
 
+  // For batch_isend_irecv, ncclGroupStart() would be called upfront
+  bool batchP2P = ncclActiveGroupCounter_ > 0;
```

Comment:
is it safe to assume ncclGroupStart() will always increment this counter?

Reply:
This counter is maintained by ProcessGroupNCCL instead of NCCL, here: https://github.com/pytorch/pytorch/blob/5f923fed6fcb7f0161395892c0a4310c88fbb5c4/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp#L2231